File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        d={'a','e','i','o','u'}
        ans=-(1<<31)
        cnt=0
        for i in range(k):
            if s[i] in d:
                cnt=cnt+1
        ans=max(ans,cnt) 
        for i in range(k,len(s)):
            if s[i-k] in d:
                cnt=cnt-1
            if s[i] in d:
                cnt=cnt+1
            ans=max(ans,cnt) 
        return ans

        

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        if(s.length()<k){
            return 0;
        }
        HashSet<Character> hs=new HashSet<>();
        hs.add('a');
        hs.add('e');
        hs.add('i');
        hs.add('o');
        hs.add('u');
        int maxi=0;
        int countVowels=0;
        for(int i=0;i<k;i++){
            if(hs.contains(s.charAt(i))){
                countVowels++;
            }
        }
        maxi=Math.max(maxi,countVowels);
        for(int i=k;i<s.length();i++){
            if(hs.contains(s.charAt(i-k))){
                countVowels--;
            }
            if(hs.contains(s.charAt(i))){
                countVowels++;
            }
            maxi=Math.max(maxi,countVowels);
        }
        return maxi;
    }
}